update: describe why all members are @Deprecated in DefaultImpls (#3159)
JetBrains/kotlin-web-site: docs/topics/jvm/java-to-kotlin-interop.md

#### all-compatibility {initial-collapse-state="collapsed"}

In addition to the `all` mode, generate compatibility stubs in the [`DefaultImpls`](https://blog.jetbrains.com/kotlin/2020/07/kotlin-1-4-m3-generating-default-methods-in-interfaces/)
classes. Compatibility stubs could be useful for 
library and runtime authors to keep backward binary compatibility for existing clients compiled against previous library versions. 
`all` and `all-compatibility` modes are changing the library ABI surface that clients will use after the recompilation of the library. 
In that sense, clients might be incompatible with previous library versions. 
This usually means that you need a proper library versioning, for example, major version increase in SemVer.

> All the members of generated `DefaultImpls` are marked with the `@Deprecated` annotation. This is to remind  that these 

Author: ```suggestion
All the members of generated `DefaultImpls` are marked with the `@Deprecated` annotation. This is to remind you that you shouldn't use these 
```

> The compiler generates all the members of `DefaultImpls` with the `@Deprecated` annotation. This is to remind you that you shouldn't use these 

Reviewer: I found "This is" too synthetic.
Let's move in the direction when one sentence comes right after the other smoothly, without any special words.
For example, we could omit "Thi is to remind you" part:

> The compiler generates all the members of `DefaultImpls` with the `@Deprecated` annotation: you shouldn't use these 
members in Java code, because the compiler generates them only for compatibility purposes.

Author: Ok, thank you, fixed.

> The compiler generates all the members of `DefaultImpls` with the `@Deprecated` annotation. This is to remind you that you shouldn't use these 

Author: ```suggestion
The compiler generates all the members of `DefaultImpls` with the `@Deprecated` annotation: you shouldn't use these 
```

> members shouldn't be used in Java code, they are generated only for compatibility purposes.

Author: ```suggestion
members in Java code, they are generated only for compatibility purposes.
```

> members in Java code, they are generated only for compatibility purposes.

Author: ```suggestion
members in Java code, the compiler generates them only for compatibility purposes.
```

> members in Java code, the compiler generates them only for compatibility purposes.

Author: ```suggestion
members in Java code, because the compiler generates them only for compatibility purposes.
```
